# ATLETAS OLÍMPICOS (1896 - 2016)
#### Quais são os maiores campeões olímpicos? E os medalhistas? Quais países tiveram mais participações olímpicas? Análises feitas para todos os países, e também para o Brasil

### PARTE 1 - Conhecendo os dados

In [3]:
import pandas as pd
from sqlalchemy import create_engine
import pandas as pd
import os

In [4]:
csv_file = r'dataset_olympics.csv'
df = pd.read_csv(csv_file)
engine = create_engine('sqlite:///atletas_olimpicos.db')
df.to_sql('olympics_data', con=engine, index=False, if_exists='replace')

70000

### 1. Análise exploratória

#### 1.1 Primeiras linhas

In [7]:
query = "SELECT * FROM OLYMPICS_DATA LIMIT 5"
df_query = pd.read_sql(query,con=engine)

In [8]:
df_query

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball  None  
1      Judo Men's Extra-Lightweight  None  
2           Football Men's Football  None  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres  None

#### 1.2 Quantidade de linhas:

In [9]:
query = """SELECT COUNT(NAME) AS [Quantidade de linhas]
            FROM OLYMPICS_DATA"""
df_query = pd.read_sql(query,con=engine)
df_query

Quantidade de linhas
0                 70000

#### 1.2.1 Quantidade de atletas distintos

In [48]:
query = """SELECT COUNT(DISTINCT NAME) AS [Atletas Distintos]
            FROM OLYMPICS_DATA"""
df_query = pd.read_sql(query,con=engine)
df_query

Atletas Distintos
0              35556

#### 1.2.2 Quantidade de nacionalidades

In [49]:
query = """SELECT COUNT(DISTINCT TEAM) AS [Qtd. Nacionalidades]
            FROM OLYMPICS_DATA"""
df_query = pd.read_sql(query,con=engine)
df_query

Qtd. Nacionalidades
0                  827

#### 1.3 Qual a primeira olimpíada na tabela?

In [10]:
query = """ SELECT DISTINCT YEAR, SEASON, CITY 
            FROM OLYMPICS_DATA 
            ORDER BY YEAR ASC    
            LIMIT 1
        """
df_query = pd.read_sql(query,con=engine)
df_query

Year  Season    City
0  1896  Summer  Athina

#### 1.4 Qual a Olimpíada mais recente na tabela?

In [11]:
query = """ SELECT DISTINCT YEAR, SEASON, CITY 
            FROM OLYMPICS_DATA 
            ORDER BY YEAR DESC    
            LIMIT 1
        """
df_query = pd.read_sql(query,con=engine)
df_query

Year  Season            City
0  2016  Summer  Rio de Janeiro

#### 1.5 Como está o agrupamento?

##### 1.5.1 Analisando a frequência dos atletas:

In [12]:
query = """ SELECT NAME, YEAR, COUNT(NAME)
            FROM OLYMPICS_DATA 
            GROUP BY NAME, YEAR  
            ORDER BY COUNT(NAME) DESC
            LIMIT 5
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name  Year  COUNT(NAME)
0  Wilhelm (William) Hunt Diederich  1932           19
1                   Acee Blue Eagle  1932           18
2             Georges-mile Fauvelle  1928           16
3                   Pierre De Soete  1928           16
4                   Marcel Bouraine  1924           13

#### Um nome aparece em 19 linhas, vamos visualizá-lo:

In [13]:
query = """ SELECT *
            FROM OLYMPICS_DATA 
            WHERE NAME = 'Wilhelm (William) Hunt Diederich'
        """
df_query = pd.read_sql(query,con=engine)
df_query

ID                              Name Sex   Age Height Weight  \
0   28407  Wilhelm (William) Hunt Diederich   M  44.0   None   None   
1   28407  Wilhelm (William) Hunt Diederich   M  44.0   None   None   
2   28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
3   28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
4   28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
5   28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
6   28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
7   28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
8   28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
9   28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
10  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
11  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
12  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
13  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
14  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
15  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
16  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
17  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
18  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
19  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   
20  28407  Wilhelm (William) Hunt Diederich   M  48.0   None   None   

             Team  NOC        Games  Year  Season         City  \
0   United States  USA  1928 Summer  1928  Summer    Amsterdam   
1   United States  USA  1928 Summer  1928  Summer    Amsterdam   
2   United States  USA  1932 Summer  1932  Summer  Los Angeles   
3   United States  USA  1932 Summer  1932  Summer  Los Angeles   
4   United States  USA  1932 Summer  1932  Summer  Los Angeles   
5   United States  USA  1932 Summer  1932  Summer  Los Angeles   
6   United States  USA  1932 Summer  1932  Summer  Los Angeles   
7   United States  USA  1932 Summer  1932  Summer  Los Angeles   
8   United States  USA  1932 Summer  1932  Summer  Los Angeles   
9   United States  USA  1932 Summer  1932  Summer  Los Angeles   
10  United States  USA  1932 Summer  1932  Summer  Los Angeles   
11  United States  USA  1932 Summer  1932  Summer  Los Angeles   
12  United States  USA  1932 Summer  1932  Summer  Los Angeles   
13  United States  USA  1932 Summer  1932  Summer  Los Angeles   
14  United States  USA  1932 Summer  1932  Summer  Los Angeles   
15  United States  USA  1932 Summer  1932  Summer  Los Angeles   
16  United States  USA  1932 Summer  1932  Summer  Los Angeles   
17  United States  USA  1932 Summer  1932  Summer  Los Angeles   
18  United States  USA  1932 Summer  1932  Summer  Los Angeles   
19  United States  USA  1932 Summer  1932  Summer  Los Angeles   
20  United States  USA  1932 Summer  1932  Summer  Los Angeles   

               Sport                                              Event Medal  
0   Art Competitions         Art Competitions Mixed Painting, Paintings  None  
1   Art Competitions        Art Competitions Mixed Sculpturing, Statues  None  
2   Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
3   Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
4   Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
5   Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
6   Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
7   Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
8   Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
9   Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
10  Art Competitions     Art Competitions Mixed Painting, Unknown Event  None  
11  Art Competitions     Art Co

#### Esse atleta apareceu em 19 linhas na tabela e participou de 2 Olimpíadas, cada categoria competida em uma linha

#### 1.6 Qual a quantidade de linhas referentes à Olimpíada de Verão e de Inverno? 

In [6]:
query = """ SELECT SEASON, COUNT(SEASON) AS [Qtd. edições]
            FROM OLYMPICS_DATA 
            GROUP BY SEASON
        """
df_query = pd.read_sql(query,con=engine)
df_query

Season  Qtd. edições
0  Summer         58467
1  Winter         11533

__________________________________________________________________________________________________________________________

### CONCLUSÕES - PARTE 1

- Tabela com 70.000 linhas, 35,5k atletas distintos e 827 nacionalidades distintas. Cada linha representa a participação de um atleta em uma determinada categoria. Por exemplo, se um atleta disputou, numa mesma edição, 100m livres e 50m costas, cada uma dessas categorias vai estar numa linha na tabela.
- A primeira Olimpíada da tabela foi em Atenas, 1896, e a mais recente foi no Rio de Janeiro em 2016
- 58.467 das 70.000 linhas são de Olimpíadas de Verão, que serão nosso foco

__________________________________________________________________________________________________________________________

### 2. Análise Geral dos Dados de Olimpíadas de Verão - Países com mais participações, maiores campeões olímpicos, etc

#### 2.1 Quais países tiveram mais participações em Olimpíadas? 
##### Exemplo, um atleta que disputou barras e solo na ginástica numa mesma edição, está 2x na tabela e conta duas vezes nessa seleção

In [8]:
query = """ SELECT TEAM, COUNT(TEAM) as [Qtd. Participações]
            FROM OLYMPICS_DATA 
            WHERE SEASON = 'Summer'
            GROUP BY TEAM
            ORDER BY COUNT(TEAM) DESC
            LIMIT 11
        """
df_query = pd.read_sql(query,con=engine)
df_query

Team  Qtd. Participações
0   United States                4044
1          France                3863
2           Italy                3209
3   Great Britain                2932
4       Australia                2030
5          Canada                1849
6           Spain                1729
7          Sweden                1508
8          Brazil                1508
9         Germany                1483
10        Hungary                1382

#### 2.1.1 Em quais posições estão Rússia/URSS, China e Alemanha (unificada, Oriental e Ocidental)?

In [15]:
query = """ SELECT 
                CASE WHEN TEAM LIKE '%Russia%' THEN 'Russia'
                     WHEN TEAM LIKE '%SOV%' THEN 'URSS'
                     WHEN TEAM LIKE '%China%' THEN 'China'
                     WHEN TEAM LIKE 'Germany%' THEN 'Germany'
                ELSE TEAM
                END AS TEAM,
                COUNT(TEAM) [Qtd. Participações]
            FROM OLYMPICS_DATA 
            WHERE   SEASON = 'Summer'
                    AND (TEAM LIKE '%SOV%'
                            OR TEAM LIKE '%Russia%'
                            OR TEAM LIKE '%China%'
                            OR TEAM LIKE 'Germany%')
            GROUP BY CASE WHEN TEAM LIKE '%Russia%' THEN 'Russia'
                          WHEN TEAM LIKE '%SOV%' THEN 'URSS'
                          WHEN TEAM LIKE '%CHINA%' THEN 'China'  
                          WHEN TEAM LIKE 'Germany%' THEN 'Germany'
                ELSE TEAM
                END
            ORDER BY COUNT(TEAM) DESC
        """
df_query = pd.read_sql(query,con=engine)
df_query

TEAM  Qtd. Participações
0  Germany                1499
1     URSS                 976
2   Russia                 717
3    China                 542

#### Rússia e URSS somadas as tornariam a 8ª nação com mais participações da história, com 1.693 participações
#### A Alemanha com todas as participações antes da unificação não trocaria de lugar
#### A China é a 26ª nação com mais participações em modalidades

#### 2.2 Quais países tiveram mais atletas distintos nas Olimpíadas?
##### Exemplo, um mesmo atleta que disputou barras e solo na ginástica numa mesma edição, está 2x na tabela e conta UMA vez nessa seleção

In [17]:
query = """ SELECT TEAM, COUNT(DISTINCT NAME) AS [Atletas Distintos]
            FROM OLYMPICS_DATA 
            WHERE SEASON = 'Summer'
            GROUP BY TEAM
            ORDER BY COUNT(DISTINCT NAME) DESC
            LIMIT 11
        """
df_query = pd.read_sql(query,con=engine)
df_query

Team  Atletas Distintos
0   United States               2164
1          France               1946
2   Great Britain               1558
3           Italy               1545
4       Australia               1057
5          Canada                974
6           Spain                873
7          Brazil                794
8         Germany                750
9          Sweden                726
10    Netherlands                670

#### Não há grandes mudanças quando quantificamos por atletas distintos em relação ao total de participações por categorias

#### 2.3 Quais atletas foram campeões olímpicos mais vezes? 

In [18]:
query = """ SELECT NAME, TEAM, SEX, SPORT, 
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT YEAR) AS [Participações],
                COUNT(MEDAL) AS [Qtd. ouros]
            FROM OLYMPICS_DATA 
            WHERE MEDAL='Gold' AND SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT
            ORDER BY COUNT(MEDAL) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name            Team Sex       Sport  \
0     Raymond Clarence "Ray" Ewry   United States   M   Athletics   
1  Matthew Nicholas "Matt" Biondi   United States   M    Swimming   
2              Usain St. Leo Bolt         Jamaica   M   Athletics   
3    Nikolay Yefimovich Andrianov    Soviet Union   M  Gymnastics   
4       Viktor Ivanovych Chukarin    Soviet Union   M  Gymnastics   
5          Vra slavsk (-Odloilov)  Czechoslovakia   F  Gymnastics   
6          Allyson Michelle Felix   United States   F   Athletics   

   Primeira Olimpíada  Última Olimpíada  Participações  Qtd. ouros  
0                1900              1908              4          10  
1                1984              1992              3           8  
2                2008              2016              3           8  
3                1972              1980              3           7  
4                1952              1956              2           7  
5                1964              1968              2           7  
6                2008              2016              3           6

#### Filtrando o Raymond, maior campeão olímpico de todos os tempos:

In [19]:
query = """ SELECT *
            FROM OLYMPICS_DATA 
            WHERE NAME = 'Raymond Clarence "Ray" Ewry' 
                AND SEASON = 'Summer'
        """
df_query = pd.read_sql(query,con=engine)
df_query

ID                         Name Sex   Age  Height  Weight  \
0  33557  Raymond Clarence "Ray" Ewry   M  26.0   185.0    79.0   
1  33557  Raymond Clarence "Ray" Ewry   M  26.0   185.0    79.0   
2  33557  Raymond Clarence "Ray" Ewry   M  26.0   185.0    79.0   
3  33557  Raymond Clarence "Ray" Ewry   M  30.0   185.0    79.0   
4  33557  Raymond Clarence "Ray" Ewry   M  30.0   185.0    79.0   
5  33557  Raymond Clarence "Ray" Ewry   M  30.0   185.0    79.0   
6  33557  Raymond Clarence "Ray" Ewry   M  32.0   185.0    79.0   
7  33557  Raymond Clarence "Ray" Ewry   M  32.0   185.0    79.0   
8  33557  Raymond Clarence "Ray" Ewry   M  34.0   185.0    79.0   
9  33557  Raymond Clarence "Ray" Ewry   M  34.0   185.0    79.0   

            Team  NOC        Games  Year  Season       City      Sport  \
0  United States  USA  1900 Summer  1900  Summer      Paris  Athletics   
1  United States  USA  1900 Summer  1900  Summer      Paris  Athletics   
2  United States  USA  1900 Summer  1900  Summer      Paris  Athletics   
3  United States  USA  1904 Summer  1904  Summer  St. Louis  Athletics   
4  United States  USA  1904 Summer  1904  Summer  St. Louis  Athletics   
5  United States  USA  1904 Summer  1904  Summer  St. Louis  Athletics   
6  United States  USA  1906 Summer  1906  Summer     Athina  Athletics   
7  United States  USA  1906 Summer  1906  Summer     Athina  Athletics   
8  United States  USA  1908 Summer  1908  Summer     London  Athletics   
9  United States  USA  1908 Summer  1908  Summer     London  Athletics   

                                  Event Medal  
0    Athletics Men's Standing High Jump  Gold  
1    Athletics Men's Standing Long Jump  Gold  
2  Athletics Men's Standing Triple Jump  Gold  
3    Athletics Men's Standing High Jump  Gold  
4    Athletics Men's Standing Long Jump  Gold  
5  Athletics Men's Standing Triple Jump  Gold  
6    Athletics Men's Standing High Jump  Gold  
7    Athletics Men's Standing Long Jump  Gold  
8    Athletics Men's Standing High Jump  Gold  
9    Athletics Men's Standing Long Jump  Gold

#### Ele participou de 4 edições entre 1900 e 1908 (1900, 1904, 1906 e 1908), ganhando todas as provas que participou (4 em salto em altura, 4 em salto em distância e 2 em salto triplo)

#### Agora, sobre aproveitamento de medalhas sobre participações (seja por Olimpíada participada, ou por modalidade competida)

#### 2.4.1 Quais atletas tiveram melhor aproveitamento de medalhas de ouro comparadas às vezes que competiram?

In [20]:
query = """ SELECT NAME, TEAM, SEX, SPORT, EVENT,
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT YEAR) AS [Edições Participadas],
                COUNT(MEDAL) AS [Qtd. ouros],
                COUNT(EVENT) AS [Vezes que competiu],
                COUNT(MEDAL)/COUNT(EVENT) AS [Ouros/Disputas]                
            FROM OLYMPICS_DATA 
            WHERE MEDAL='Gold'
                    AND SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT, EVENT
            ORDER BY COUNT(MEDAL)/COUNT(EVENT) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name  \
0                      A. Albert   
1  Aage Jrgen Christian Andersen   
2          Aaron Nigel Armstrong   
3                 Abas Arslanagi   
4          Abdul Rashid Al-Hasan   
5                   Abebe Bikila   
6        Abelardo Fernndez Antua   

                                             Team Sex      Sport  \
0  Union des Socits Franais de Sports Athletiques   M      Rugby   
1                                         Denmark   M   Football   
2                             Trinidad and Tobago   M  Athletics   
3                                      Yugoslavia   M   Handball   
4                                        Pakistan   M     Hockey   
5                                        Ethiopia   M  Athletics   
6                                           Spain   M   Football   

                                  Event  Primeira Olimpíada  Última Olimpíada  \
0                     Rugby Men's Rugby                1900              1900   
1               Football Men's Football                1906              1906   
2  Athletics Men's 4 x 100 metres Relay                2008              2008   
3               Handball Men's Handball                1972              1972   
4                   Hockey Men's Hockey                1984              1984   
5              Athletics Men's Marathon                1960              1964   
6               Football Men's Football                1992              1992   

   Edições Participadas  Qtd. ouros  Vezes que competiu  Ouros/Disputas  
0                     1           1                   1               1  
1                     1           1                   1               1  
2                     1           1                   1               1  
3                     1           1                   1               1  
4                     1           1                   1               1  
5                     2           2                   2               1  
6                     1           1                   1               1

#### 2.4.2 Há vários atletas que participaram só uma vez. Vamos calcular a mesma coisa, porém para atletas que mais participaram em modalidades em Olimpíadas

In [21]:
query = """ SELECT NAME, TEAM, SEX, SPORT,
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT YEAR) AS [Edições Participadas],
                COUNT(MEDAL) AS [Qtd. ouros],
                COUNT(EVENT) AS [Vezes que competiu],
                COUNT(MEDAL)/COUNT(EVENT) AS [Ouros/Disputas]                
            FROM OLYMPICS_DATA 
            WHERE MEDAL='Gold'
                    AND SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT
            ORDER BY COUNT(EVENT) DESC, COUNT(MEDAL)/COUNT(EVENT) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name            Team Sex       Sport  \
0     Raymond Clarence "Ray" Ewry   United States   M   Athletics   
1  Matthew Nicholas "Matt" Biondi   United States   M    Swimming   
2              Usain St. Leo Bolt         Jamaica   M   Athletics   
3    Nikolay Yefimovich Andrianov    Soviet Union   M  Gymnastics   
4       Viktor Ivanovych Chukarin    Soviet Union   M  Gymnastics   
5          Vra slavsk (-Odloilov)  Czechoslovakia   F  Gymnastics   
6          Allyson Michelle Felix   United States   F   Athletics   

   Primeira Olimpíada  Última Olimpíada  Edições Participadas  Qtd. ouros  \
0                1900              1908                     4          10   
1                1984              1992                     3           8   
2                2008              2016                     3           8   
3                1972              1980                     3           7   
4                1952              1956                     2           7   
5                1964              1968                     2           7   
6                2008              2016                     3           6   

   Vezes que competiu  Ouros/Disputas  
0                  10               1  
1                   8               1  
2                   8               1  
3                   7               1  
4                   7               1  
5                   7               1  
6                   6               1

#### 2.4.2 Quais atletas tiveram maior média de ouros por edições olímpicas que participaram?

In [22]:
query = """ SELECT NAME, TEAM, SEX, SPORT, 
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT YEAR) AS [Participações],
                COUNT(MEDAL) AS [Qtd. ouros],
                COUNT(MEDAL)/COUNT(DISTINCT YEAR) AS [Ouros/Edições Participadas]                
            FROM OLYMPICS_DATA 
            WHERE MEDAL='Gold'
                    AND SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT
            ORDER BY COUNT(MEDAL)/COUNT(DISTINCT YEAR) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name           Team Sex       Sport  \
0  Francina Elsje "Fanny" Blankers-Koen    Netherlands   F   Athletics   
1    Kornelia Ender (-Matthes, -Grummt)   East Germany   F    Swimming   
2                  Simone Arianne Biles  United States   F  Gymnastics   
3        Vladimir Nikolayevich Artyomov   Soviet Union   M  Gymnastics   
4       Aleksandr Nikolayevich Dityatin   Soviet Union   M  Gymnastics   
5                         Dennis Fenton  United States   M    Shooting   
6      Dmitry Vladimirovich Bilozerchev   Soviet Union   M  Gymnastics   

   Primeira Olimpíada  Última Olimpíada  Participações  Qtd. ouros  \
0                1948              1948              1           4   
1                1976              1976              1           4   
2                2016              2016              1           4   
3                1988              1988              1           4   
4                1980              1980              1           3   
5                1920              1920              1           3   
6                1988              1988              1           3   

   Ouros/Edições Participadas  
0                           4  
1                           4  
2                           4  
3                           4  
4                           3  
5                           3  
6                           3

#### 2.5 Quais os maiores medalhistas olímpicos?

In [23]:
query = """ SELECT NAME, TEAM, SEX, SPORT, 
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT YEAR) AS [Participações],
                SUM(CASE WHEN MEDAL = 'Gold' THEN 1 ELSE 0 END) AS [Qtd. Ouro],
                SUM(CASE WHEN MEDAL = 'Silver' THEN 1 ELSE 0 END) AS [Qtd. Prata],
                SUM(CASE WHEN MEDAL = 'Bronze' THEN 1 ELSE 0 END) AS [Qtd. Bronze],
                COUNT(MEDAL) AS [Qtd. Medalhas]
            FROM OLYMPICS_DATA 
            WHERE SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT
            ORDER BY COUNT(MEDAL) DESC
            LIMIT 5
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name            Team Sex       Sport  \
0    Nikolay Yefimovich Andrianov    Soviet Union   M  Gymnastics   
1   Natalie Anne Coughlin (-Hall)   United States   F    Swimming   
2  Matthew Nicholas "Matt" Biondi   United States   M    Swimming   
3       Viktor Ivanovych Chukarin    Soviet Union   M  Gymnastics   
4          Vra slavsk (-Odloilov)  Czechoslovakia   F  Gymnastics   

   Primeira Olimpíada  Última Olimpíada  Participações  Qtd. Ouro  Qtd. Prata  \
0                1972              1980              3          7           5   
1                2004              2012              3          3           4   
2                1984              1992              3          8           2   
3                1952              1956              2          7           3   
4                1960              1968              3          7           4   

   Qtd. Bronze  Qtd. Medalhas  
0            3             15  
1            5             12  
2            1             11  
3            1             11  
4            0             11

#### 2.6 Quais esportes tiveram maiores médias de altura?

In [24]:
query = """ SELECT SPORT, ROUND(AVG(HEIGHT),1) AS [Média Altura]
            FROM OLYMPICS_DATA 
            WHERE SEASON='Summer'
            GROUP BY SPORT
            ORDER BY AVG(HEIGHT) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Sport  Média Altura
0        Basketball         190.8
1        Volleyball         187.3
2  Beach Volleyball         187.1
3        Water Polo         184.9
4            Rowing         184.3
5          Baseball         183.1
6          Handball         182.9

#### 2.7 Esportes com maiores aumentos de média de altura a partir de 2004 comparada com as edições até 1970

In [25]:
query = """ SELECT SPORT,
                AVG(CASE WHEN YEAR <= 1970 THEN height END) AS [Média Altura até 1970],
                AVG(CASE WHEN YEAR >= 2004 THEN height END) AS [Média Altura pós 2004],
                ROUND(((AVG(CASE WHEN YEAR >= 2004 THEN height END) - AVG(CASE WHEN YEAR <= 1970 THEN height END)) / AVG(CASE WHEN YEAR <= 1970 THEN height END)) * 100, 2) AS [Aumento da Média da Altura]
            FROM OLYMPICS_DATA 
            WHERE SEASON='Summer'
            GROUP BY SPORT
            ORDER BY ROUND(((AVG(CASE WHEN YEAR >= 2004 THEN height END) - AVG(CASE WHEN YEAR <= 1970 THEN height END)) / AVG(CASE WHEN YEAR <= 1970 THEN height END)) * 100, 2) DESC
            LIMIT 5
        """
df_query = pd.read_sql(query,con=engine)
df_query

Sport  Média Altura até 1970  Média Altura pós 2004  \
0  Volleyball             181.058824             189.846416   
1      Tennis             174.550000             180.618519   
2    Swimming             174.534398             180.297500   
3    Canoeing             176.089552             180.151220   
4  Basketball             188.229839             191.358553   

   Aumento da Média da Altura  
0                        4.85  
1                        3.48  
2                        3.30  
3                        2.31  
4                        1.66

__________________________________________________________________________________________________________________________

### CONCLUSÃO - PARTE 2 - ANÁLISE GERAL

- Contando participação em cada modalidade, os EUA tiveram mais participações, com aproximadamente 5k, seguidos pela França com 4,6k
- Contando participações por edições olímpicas, os EUA também lideram as participações, com 2,5k, seguidos pela França com 2,2k.
- Raymond Clarence "Ray" Ewry, estadunidense que disputou as 4 edições entre 1900 e 1908 (teve edição em 1906) no atletismo, foi o maior campeão da história da Olimpíada. Ele ganhou todas as 10 modalidades que disputou nessas 4 edições: 4 em salto em altura, 4 em salto em distância e 2 em salto triplo 
- Levando em conta ouros em EDIÇÕES olímpicas participadas, destaque para Simone Biles que se juntou ao seleto grupo de 4 atletas que ganharam 4 ouros numa mesma edição olímpica
- Basquete é o esporte com maior média de altura (levando em conta homens e mulheres), com 1,908m, seguido por vôlei com 1,873m
- Vôlei é o esporte que teve maior aumento de média de altura comparando atletas a partir de 2004 com atletas que participaram das edições até 1970, com aumento de 4,8%

__________________________________________________________________________________________________________________________

### 3. Participações do Brasil

##### O país teve 1.529 participações a nível de modalidade (11º geral), e 804 atletas distintos (9º geral)

#### 3.1 Quais atletas tiveram mais participações, levando em conta as modalidades?
##### Por exemplo, se um atleta disputou uma modalidade em duas categorias (50m nado livre e 100m costas), será contado 2x

In [26]:
query = """ SELECT NAME, TEAM, SEX, SPORT, 
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT EVENT) AS [Qtd. Modalidades]
            FROM OLYMPICS_DATA 
            WHERE TEAM='Brazil'
                    AND SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT
            ORDER BY COUNT(DISTINCT EVENT) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name    Team Sex          Sport  \
0  Francisco Carlos Barretto Jnior  Brazil   M     Gymnastics   
1                     Camila Comin  Brazil   F     Gymnastics   
2             Gustavo Frana Borges  Brazil   M       Swimming   
3           Jade Fernandes Barbosa  Brazil   F     Gymnastics   
4  Renyldo Pedro Guimares Ferreira  Brazil   M  Equestrianism   
5   Cludia de Paula Costa Magalhes  Brazil   F     Gymnastics   
6             Cyro Marques Delgado  Brazil   M       Swimming   

   Primeira Olimpíada  Última Olimpíada  Qtd. Modalidades  
0                2016              2016                 7  
1                2000              2004                 6  
2                1992              2004                 6  
3                2008              2016                 6  
4                1948              1960                 6  
5                1980              1980                 5  
6                1980              1984                 5

#### 3.2 Quais atletas participaram de mais edições olímpicas?
##### Por exemplo, se um atleta disputou uma modalidade em duas categorias (50m nado livre e 100m costas) numa mesma edição, será contado 1x

In [27]:
query = """ SELECT NAME, TEAM, SEX, SPORT, 
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT YEAR) AS [Qtd. Edições Olímpicas]
            FROM OLYMPICS_DATA 
            WHERE TEAM='Brazil'
                    AND SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT
            ORDER BY COUNT(DISTINCT YEAR) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name    Team Sex          Sport  \
0       Adriana "Adrianinha" Moiss Pinto  Brazil   F     Basketball   
1  Alvaro Affonso "Doda" de Miranda Neto  Brazil   M  Equestrianism   
2                  Murilo Antnio Fischer  Brazil   M        Cycling   
3                        Reinaldo Conrad  Brazil   M        Sailing   
4         Ricardo "Bimba" Winicki Santos  Brazil   M        Sailing   
5              Adhemar Ferreira da Silva  Brazil   M      Athletics   
6       Alexandra Priscila do Nascimento  Brazil   F       Handball   

   Primeira Olimpíada  Última Olimpíada  Qtd. Edições Olímpicas  
0                2000              2016                       5  
1                1996              2016                       5  
2                2000              2016                       5  
3                1960              1980                       5  
4                2000              2016                       5  
5                1948              1960                       4  
6                2004              2016                       4

#### 3.3 Quais atletas mais ganharam medalhas?

In [28]:
query = """ SELECT NAME, TEAM, SEX, SPORT, 
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT EVENT) AS [Qtd. Modalidades],
                SUM(CASE WHEN MEDAL = 'Gold' THEN 1 ELSE 0 END) AS [Qtd. Ouro],
                SUM(CASE WHEN MEDAL = 'Silver' THEN 1 ELSE 0 END) AS [Qtd. Prata],
                SUM(CASE WHEN MEDAL = 'Bronze' THEN 1 ELSE 0 END) AS [Qtd. Bronze],
                COUNT(MEDAL) AS [Qtd. Medalhas]
            FROM OLYMPICS_DATA 
            WHERE TEAM='Brazil'
                    AND SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT
            ORDER BY COUNT(MEDAL) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name    Team Sex       Sport  \
0                Gustavo Frana Borges  Brazil   M    Swimming   
1  Srgio "Escadinha" Dutra dos Santos  Brazil   M  Volleyball   
2   Bruno "Bruninho" Mossa de Rezende  Brazil   M  Volleyball   
3            Csar Augusto Cielo Filho  Brazil   M    Swimming   
4     Dante Guimares Santos do Amaral  Brazil   M  Volleyball   
5             Marcelo Bastos Ferreira  Brazil   M     Sailing   
6                 Ademir Roque Kaefer  Brazil   M    Football   

   Primeira Olimpíada  Última Olimpíada  Qtd. Modalidades  Qtd. Ouro  \
0                1992              2004                 6          0   
1                2004              2016                 1          2   
2                2008              2016                 1          1   
3                2008              2012                 3          1   
4                2000              2012                 1          1   
5                1992              2004                 2          2   
6                1984              1988                 1          0   

   Qtd. Prata  Qtd. Bronze  Qtd. Medalhas  
0           2            2              4  
1           2            0              4  
2           2            0              3  
3           0            2              3  
4           2            0              3  
5           0            1              3  
6           2            0              2

#### 3.4 Quais atletas foram mais vezes campeões olímpicos?

In [30]:
query = """ SELECT NAME, TEAM, SEX, SPORT, 
                MIN(YEAR) AS [Primeira Olimpíada],
                MAX(YEAR) AS [Última Olimpíada],
                COUNT(DISTINCT EVENT) AS [Qtd. Modalidades],
                COUNT(MEDAL) AS [Qtd. Ouros]
            FROM OLYMPICS_DATA 
            WHERE TEAM='Brazil' 
                    AND MEDAL='Gold'
                    AND SEASON='Summer'
            GROUP BY NAME, TEAM, SEX, SPORT
            ORDER BY COUNT(MEDAL) DESC
            LIMIT 7
        """
df_query = pd.read_sql(query,con=engine)
df_query

Name    Team Sex       Sport  \
0            Adhemar Ferreira da Silva  Brazil   M   Athletics   
1     Fabiana "Fabi" Alvim de Oliveira  Brazil   F  Volleyball   
2           Fabiana Marcelino Claudino  Brazil   F  Volleyball   
3              Marcelo Bastos Ferreira  Brazil   M     Sailing   
4   Srgio "Escadinha" Dutra dos Santos  Brazil   M  Volleyball   
5  Adenzia Aparecida Ferreira da Silva  Brazil   F  Volleyball   
6                       Amauri Ribeiro  Brazil   M  Volleyball   

   Primeira Olimpíada  Última Olimpíada  Qtd. Modalidades  Qtd. Ouros  
0                1952              1956                 1           2  
1                2008              2012                 1           2  
2                2008              2012                 1           2  
3                1996              2004                 2           2  
4                2004              2016                 1           2  
5                2012              2012                 1           1  
6                1992              1992                 1           1

#### 3.5 Quantos atletas ganharam ao menos uma medalha?

In [31]:
query = """ SELECT COUNT(NAME) AS [Qtd Medalhistas]
            FROM OLYMPICS_DATA 
            WHERE TEAM='Brazil' 
                AND MEDAL != 'None'
                AND SEASON = 'Summer'
        """
df_query = pd.read_sql(query,con=engine)
df_query

Qtd Medalhistas
0              163

__________________________________________________________________________________________________________________________

### CONCLUSÕES - PARTE 3 - BRASIL

- Francisco Carlos, que disputou a Olimpíada do Rio, é o atleta brasileiro com mais modalidades disputadas na Olimpíada, com 7
- Gustavo Borges e o Serginho do vôlei são os atletas com mais medalhas olímpicas, 4
- 163 atletas brasileiros já ganharam medalhas olímpicas

__________________________________________________________________________________________________________________________

### PARTE 4 - Brasil nas Olimpíadas de Inverno

### 4.1 O Brasil já ganhou medalha nas Olimpíadas de Inverno?

In [35]:
query = """ SELECT COUNT(NAME) AS [Qtd Medalhas]
            FROM OLYMPICS_DATA 
            WHERE TEAM='Brazil' 
                AND MEDAL != 'None'
                AND SEASON = 'Winter'
        """
df_query = pd.read_sql(query,con=engine)
df_query

Qtd Medalhas
0             0

### 4.2 Quais modalidades o Brasil mais participou nas Olimpíadas de Inverno?

In [37]:
query = """ SELECT EVENT, COUNT(EVENT) AS [Qtd Participações]
            FROM OLYMPICS_DATA 
            WHERE TEAM='Brazil' 
                AND SEASON = 'Winter'
        """
df_query = pd.read_sql(query,con=engine)
df_query

Event  Qtd Participações
0  Alpine Skiing Men's Super G                 21

__________________________________________________________________________________________________________________________

### CONCLUSÕES - PARTE 4 - BRASIL NAS OLIMPÍADAS DE INVERNO

- O Brasil já participou em 21 modalidades nas Olimpíadas de Inverno, sempre no ski, e nunca ganhou medalha 

__________________________________________________________________________________________________________________________

### CONCLUSÕES

#### CONCLUSÃO - PARTE 1: ANÁLISE SOBRE A BASE

- Tabela com 70.000 linhas, 35,5k atletas distintos e 827 nacionalidades distintas
- A primeira Olimpíada da tabela foi em Atenas, 1896, e a mais recente foi no Rio de Janeiro em 2016

##### CONCLUSÃO - PARTE 2: ANÁLISES GERAIS
- Contando participação em cada modalidade, os EUA tiveram mais participações, com aproximadamente 5k, seguidos pela França com 4,6k
- Contando participações por edições olímpicas, os EUA também lideram as participações, com 2,5k, seguidos pela França com 2,2k.
- Raymond Clarence "Ray" Ewry, estadunidense que disputou as 4 edições entre 1900 e 1908 (teve edição em 1906) no atletismo, foi o maior campeão da história da Olimpíada. Ele ganhou todas as 10 modalidades que disputou nessas 4 edições: 4 em salto em altura, 4 em salto em distância e 2 em salto triplo 
- Levando em conta ouros em EDIÇÕES olímpicas participadas, destaque para Simone Biles que se juntou ao seleto grupo de 4 atletas que ganharam 4 ouros numa mesma edição olímpica
- Basquete é o esporte com maior média de altura (levando em conta homens e mulheres), com 1,908m, seguido por vôlei com 1,873m
- Vôlei é o esporte que teve maior aumento de média de altura comparando atletas a partir de 2004 com atletas que participaram das edições até 1970, com aumento de 4,8%

##### CONCLUSÕES - PARTE 3: BRASIL:
- Francisco Carlos, que disputou a Olimpíada do Rio, é o atleta brasileiro com mais modalidades disputadas na Olimpíada, com 7
- Gustavo Borges e o Serginho do vôlei são os atletas com mais medalhas olímpicas, 4
- 163 atletas brasileiros já ganharam medalhas olímpicas

##### CONCLUSÕES - PARTE 4 - BRASIL NAS OLIMPÍADAS DE INVERNO:
- O Brasil já participou em 21 modalidades nas Olimpíadas de Inverno, sempre no ski, e nunca ganhou medalha

__________________________________________________________________________________________________________________________